.github/workflows/warn-on-editing-requirements.yml

```diff
@@ -0,0 +1,43 @@
+name: Maintainer approval for requirement edits
+
+on:
+  pull_request_review:
+    types: [submitted]
+
+permissions:
+  pull-requests: read
+
+jobs:
+  check:
+    if: github.event.review.state == 'APPROVED'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Fail without core maintainer approval
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const prNumber = context.payload.pull_request.number;
```

Comment:
can we move the code to the js file?

Reply:
I see, we might use require-core-maintainer-approval.js for a reason in the future

Reply:
ah yes that was my intention